### PR TITLE
[Blog Post] Resources - Migrating Your React Native Library to the New Architecture

### PR DESCRIPTION
**Pull request by cipolleschi**
This PR introduce a Blog Post on the React Native Website related to all the work we did to support library developer for React Native in migrating their code to the New Architecture.

### CODE REVIEW

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+title: Resources - Migrating Your React Native Library to the New Architecture
+author: "cipolleschi"
+tags: [announcement]
+date: 2022-06-16
+---
+
+**TL; DR**: We are working on improving the resources supporting the react Native New Architecture. We already released a repository to help migrating your app ([RNNewArchitectureApp](https://github.com/react-native-community/RNNewArchitectureApp)) and one for your libraries ([RNNewArchitectureLibraries](https://github.com/react-native-community/RNNewArchitectureLibraries)). We are also revamping the [New Architecture guide](https://github.com/facebook/react-native-website/pull/3037) on the Website and we created a [GitHub Working Group](https://github.com/reactwg/react-native-new-architecture/discussions) to answer questions related to the new Architecture.
```
Comment (cortinico) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:8]:
```suggestion
**tl;dr**: We are working on improving the resources supporting the React Native New Architecture. We already released a repository to help migrating your app ([RNNewArchitectureApp](https://github.com/react-native-community/RNNewArchitectureApp)) and one for your libraries ([RNNewArchitectureLibraries](https://github.com/react-native-community/RNNewArchitectureLibraries)). We are also revamping the [New Architecture guide](https://github.com/facebook/react-native-website/pull/3037) on the Website and we created a [GitHub Working Group](https://github.com/reactwg/react-native-new-architecture/discussions) to answer questions related to the New Architecture.
```

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+---
+title: Resources - Migrating Your React Native Library to the New Architecture
+author: "cipolleschi"
```
Comment (cortinico) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:3]: @cipolleschi You need to add yourself to `website/blog/authors.yml`

**Review by cortinico — APPROVED** [review]
LGTM. Minor nits 👍

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md (hunk 2 above)
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md]: thanks

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+---
+title: Resources - Migrating Your React Native Library to the New Architecture
+author: 'cipolleschi'
+tags: [announcement]
+date: 2022-06-16
+---
+
+**tl; dr**: We are working on improving the resources supporting the react Native New Architecture. We already released a repository to help migrating your app ([RNNewArchitectureApp](https://github.com/react-native-community/RNNewArchitectureApp)) and one for your libraries ([RNNewArchitectureLibraries](https://github.com/react-native-community/RNNewArchitectureLibraries)). We are also revamping the [New Architecture guide](https://github.com/facebook/react-native-website/pull/3037) on the Website and we created a [GitHub Working Group](https://github.com/reactwg/react-native-new-architecture/discussions) to answer questions related to the new Architecture.
+
```
Comment (Simek) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:9]:
```suggestion

<!--truncate-->

```
We should add truncation marker somewhere, since the post is quite long. Not sure if it's an ideal spot, maybe we can leave also a bit of Intro.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+- **Support:** Earlier this year, we created a [GitHub Working Group](https://github.com/reactwg/react-native-new-architecture/discussions) dedicated to discussion and questions around the New Architecture.
+
+In this post, we will dig deeper into these resources and explain in more detail how you can use them most efficiently. Finally, we will provide a snapshot of the current migration state for the most used React Native libraries.
+
+### Documentation
+
+In the past 6 months, we’ve added a guide on adopting the New Architecture (https://reactnative.dev/docs/new-architecture-intro) and an architecture deep-dive (https://reactnative.dev/architecture/overview) on Fabric. We plan to expand this to include more guides and documentation around creating TurboModules, understanding CodeGen, and more. We hope to have updates to share by the 0.70 release.
```
Comment (Simek) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:26]:
```suggestion
In the past 6 months, we’ve added a guide on adopting the New Architecture (/docs/new-architecture-intro) and an architecture deep-dive (/architecture/overview) on Fabric. We plan to expand this to include more guides and documentation around creating TurboModules, understanding CodeGen, and more. We hope to have updates to share by the 0.70 release.
```

Let's use internal links for doc pages. Those prevents the whole app rerender on navigation which lead to more seamless and quicker page loads.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+At the moment, the **New Architecture** guide covers how to migrate your app (https://reactnative.dev/docs/new-architecture-app-intro) and your libraries (https://reactnative.dev/docs/new-architecture-library-intro) to support the New Architecture properly.
```
Comment (Simek) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:28]:
```suggestion
At the moment, the **New Architecture** guide covers how to migrate your app (/docs/new-architecture-app-intro) and your libraries (/docs/new-architecture-library-intro) to support the New Architecture properly.
```

Same as above.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md (hunk 4 above)
Comment (Simek) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:26]: It also would be nice to add links to the certain words instead of listing the whole URL in content.
Screenshot 2022-06-17 at 09 24 04

**Review by Simek — APPROVED** [review]
I have left a few small suggestion, but generally this LGTM! 

Thank you so much for adding this summary post! 🙏 
 
One more thing, please do not forget to fix the lint before the merge. At this time it complains about cases:
Screenshot 2022-06-17 at 09 24 59

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
+---
+title: Resources - Migrating Your React Native Library to the New Architecture
```
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:2]:
```suggestion
title: Helping migrate React Native libraries to the New Architecture
```
this title is more aligned with the tone, also I'm not a big fan of the "Resources -" at the start, feels like it should almost be a tag instead

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+author: 'cipolleschi'
+tags: [announcement]
+date: 2022-06-16
+---
+
+**tl; dr**: We are working on improving the resources supporting the react Native New Architecture. We already released a repository to help migrating your app ([RNNewArchitectureApp](https://github.com/react-native-community/RNNewArchitectureApp)) and one for your libraries ([RNNewArchitectureLibraries](https://github.com/react-native-community/RNNewArchitectureLibraries)). We are also revamping the [New Architecture guide](https://github.com/facebook/react-native-website/pull/3037) on the Website and we created a [GitHub Working Group](https://github.com/reactwg/react-native-new-architecture/discussions) to answer questions related to the new Architecture.
```
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:8]:
```suggestion
**tl; dr**: We are working on improving the resources supporting the React Native New Architecture. We already released a repository to help migrating your app ([RNNewArchitectureApp](https://github.com/react-native-community/RNNewArchitectureApp)) and one for your libraries ([RNNewArchitectureLibraries](https://github.com/react-native-community/RNNewArchitectureLibraries)). We are also revamping the [New Architecture guide](https://github.com/facebook/react-native-website/pull/3037) on the Website and we created a [GitHub Working Group](https://github.com/reactwg/react-native-new-architecture/discussions) to answer questions related to the new Architecture.
```

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md (hunk 5 above)
Comment (kelset) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:28]:
```suggestion
Currently, the **New Architecture** guide covers how to migrate your app (https://reactnative.dev/docs/new-architecture-app-intro) and your libraries (https://reactnative.dev/docs/new-architecture-library-intro) to support the New Architecture properly.
```
(nit: "at the moment" is an italian expression ;)

Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:28]:
```suggestion
Currently, the **New Architecture** guide covers how to migrate your app (https://reactnative.dev/docs/new-architecture-app-intro) and your libraries (https://reactnative.dev/docs/new-architecture-library-intro) to support the New Architecture properly.
```
nit: "at the moment" is mostly an italian expression ;)

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+#### RNNewArchitectureApp
+
+This repo was created to demonstrate how to migrate an app, the native modules and the native components from the legacy architecture (0.67) to the most recent version React Native. Each commit corresponds to an isolated migration step.
+
+<figure>
+    <img src="/blog/assets/new-arch-example-steps-to-migrate-an-app.png" alt="Example steps to migrate an app" />
+    <figcaption>Commit list for a migration in the RNNewArchitectureApp repository</figcaption>
+</figure>
+[Image: ]
```
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:44]:
```suggestion
```
leftover?

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+- [Documentation](https://github.com/reactwg/react-native-new-architecture/discussions/categories/documentation): A place to chat about the New Architecture documentation and migration material
+- [Libraries](https://github.com/reactwg/react-native-new-architecture/discussions/categories/libraries): A place to chat about 3rd party libraries and their migration story to the New Architecture
+- [Q&A](https://github.com/reactwg/react-native-new-architecture/discussions/categories/q-a): A place to ask the community for help on the New Architecture topics
+- [Releases](https://github.com/reactwg/react-native-new-architecture/discussions/categories/releases): A place to chat about release specific bugs & build problems
+
+To use this group effectively:
+
+- **Make sure your library is listed inside the [Libraries] (https://github.com/reactwg/react-native-new-architecture/discussions/categories/libraries)section**. This will help us share a status update on the migration of your library and will help us understand which struggles library authors are facing to support you better.
```
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:80]:
```suggestion
- **Make sure your library is listed inside the [Libraries](https://github.com/reactwg/react-native-new-architecture/discussions/categories/libraries) section**. This will help us share a status update on the migration of your library and will help us understand which struggles library authors are facing to support you better.
```
fix

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+- **Leverage the Q&A [section](https://github.com/reactwg/react-native-new-architecture/discussions/categories/q-a) if you face a blocker and need support**. Our team and community experts are monitoring and will support at our best effort.
+- **Keep an eye on the other sections for topics that may affect you**. A new release may introduce exactly the API that you were looking for. You can subscribe to particular discussions via Github.
+
+We plan to support this group until the **New Architecture** is enabled by default and all the major libraries have been migrated to it.
+
+### Migration Status of Certain Libraries
```
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:86]:
```suggestion
### Migration Status of Popular Libraries
```

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+
+We are invested in supporting the React Native community’s adoption of the new architecture. Concretely, we will continue to:
+
+- Offer best-effort support in the **Working Group.**
+- Provide more examples about how to achieve amazing results with the new architecture in the **RNNewArchitecture** repositories.
+- Provide clear and up-to-date documentation on the **New Architecture**.
+- Track the migration status of essential React Native libraries in the **Working Group**.
+- Simplify the migration path for developers
+
+In addition, React Native 0.69 will ship with improved devX for app and library developers for new architecture adoption. You can find more information about the 0.69.0 release [here](https://github.com/reactwg/react-native-releases/discussions/21).
```
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:110]: This phrase imply this blogpost will come out before 0.69.0 - @cortinico @fortmarek we should probably revise the blogpost draft to account for this blogpost existing (maybe?)

Personally I think we should avoid publishing two blogposts back-to-back, it makes it more likely that folks will miss out on one or the other; so if we want them to go out in the span of a few days, let's make sure that the other one has a link to the previous to raise awareness

Comment (fortmarek) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:110]: Thanks for the callout @kelset - it'd be great to link to this from the 0.69 blogpost. If we could merge this by Monday and then ~ Wednesday release 0.69, that'd be a reasonable schedule imho. @cipolleschi wdyt? I don't want to rush this if you think this needs more time.

**Review by fortmarek — COMMENTED** [review]
Great overview, just a couple of minor comments from my side 🙂

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+
+At the moment, the **New Architecture** guide covers how to migrate your app (https://reactnative.dev/docs/new-architecture-app-intro) and your libraries (https://reactnative.dev/docs/new-architecture-library-intro) to support the New Architecture properly.
+
+If you are interested in the evolution of this guide, or have feedback, you can follow along on this pull request (https://github.com/facebook/react-native-website/pull/3037).
+
+### Example Migrations
+
+For developers who may want to follow along in code, we’ve prepared two example repositories.
+
+#### RNNewArchitectureApp
```
[the remark this answers is not in the document]
Comment (Simek) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:36]: Generally adding links to the headers is something that we want to avoid, since they will always have the `#` link anyway and it might be confusing for the users.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+The repo is organized as follows:
+
+- A **main** branch has no code but a README.md which advertises other branches.
+- Several migration branches which show a migration from a specific version of RN to another.
+
+Some of the migration branches also have a **RUN.md** file which describes in a more human-readable fashion the exact steps that have been applied in every commit.
+
+We plan to keep this example up to date with the most recent stable releases, adding migrations to any minor release of React Native we are going to release. If you notice issue with any of the steps, please file an issue in the repository. This will hold until we have the reasonable feeling that most of the React Native users have migrated to the New Architecture.
+
+#### RNNewArchitectureLibraries
```
[the remark this answers is not in the document]
Comment (fortmarek) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:54]: As mentioned [here](https://github.com/facebook/react-native-website/pull/3163#discussion_r900062313), we don't want links in the headers.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md (hunk 4 above)
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:26]: Thanks for the explanation about why it is useful to use internal links!
I missed these two spots for use words for links! 😱

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md (hunk 5 above)
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:28]: Aw, thanks! :D

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md (hunk 11 above)
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:110]: Yes, sure! 

I'm apply all the comments and suggestions right now. On Monday I'll be on vacation (I had already planned to try and publish this ASAP!), but if everything is right, feel free to push the merge button!

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+## Introduction
+
+In this post we are sharing an update on tools and resources to help React Native library developers to migrate their **Native Modules** and **Native Components** to their **New Architecture** equivalents, **TurboModule** and **Fabric Components**.
+
+React Native users leverage a vast number of open-source libraries for their apps. For a complete and consistent ecosystem, it is necessary that these libraries migrate such that everyone can benefit from the unlocked capabilities and performance improvements of the new architecture.
```
[the remark this answers is not in the document]
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:14]: Thanks! I swooped the document again looking for all the `new architecture`, `new Architecture` and `New architecture` combos and now it should be consistent! 😉

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:36]: Yup, added to the `This repo` fragment in the first sentence!

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+
+If you are interested in the evolution of this guide, or have feedback, you can follow along on this pull request (https://github.com/facebook/react-native-website/pull/3037).
+
+### Example Migrations
+
+For developers who may want to follow along in code, we’ve prepared two example repositories.
+
+#### RNNewArchitectureApp
+
+This repo was created to demonstrate how to migrate an app, the native modules and the native components from the legacy architecture (0.67) to the most recent version React Native. Each commit corresponds to an isolated migration step.
```
[the remark this answers is not in the document]
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:38]: Amazing, thank you so much!

**Review by fortmarek — APPROVED** [review]
Looks good to me 🙌

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+
+<!--truncate-->
+
+## Introduction
+
+In this post we are sharing an update on tools and resources to help you migrate your **Native Modules** and **Native Components** to their **New Architecture** equivalents, **TurboModule** and **Fabric Components**.
+
+React native users leverage vast number of open source libraries for building apps. For a complete and consistent ecosystem, it is necessary that these libraries migrate such that everyone can benefit from the unlocked capabilities and performance improvements of the New Architecture.
```
Comment (Simek) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:16]:
```suggestion
React Native users leverage vast number of open source libraries for building apps. For a complete and consistent ecosystem, it is necessary that these libraries migrate such that everyone can benefit from the unlocked capabilities and performance improvements of the New Architecture.
```

Case fix, on which CI is currently failing on.
